### PR TITLE
with_prefix macro - Visibility qualifier for generated module

### LINKED ISSUE #327: with_prefix! Making the module public
On a few occasions we have the same struct with the same prefix used in our applications. Currently the `with_prefix` creates a private module for serialize/deserialize and we have to do a hack in order to use the same prefix in other places  and applications.

I'm opening this issue to see how this could be improved in future version.

```rust
// re-export the prefix module!
pub mod prefix_active {
    use serde_with::with_prefix;

    pub use prefix_active::*;

    with_prefix!(prefix_active "active_");
}
```

PS: Thank you all very much for creating and maintaining this crate!

### PR DESCRIPTION
**Pull request by elpiel**
Closes #327 

I'm not sure how to document this and what is the preferred way to add the tests, so I decided to open a PR directly

### CODE REVIEW

File: src/with_prefix.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@ use std::fmt;
 /// Serialize with an added prefix on every field name and deserialize by
 /// trimming away the prefix.
 ///
+/// You can set the visibility of the generated module by prefix the module name with a module visibility.
```
Comment (jonasbb) [inline src/with_prefix.rs:11]: Maybe you could show an example like `with_prefix!(pub(crate) prefix_foo "foo_");` and mention that it is optional.

**Review by jonasbb — CHANGES_REQUESTED** [review]

**Comment by jonasbb** [comment]
Maybe mention how to use the modifier and that it is optional. Otherwise, the PR looks good. Thanks for creating the PR.

bors delegate=elpiel

File: src/with_prefix.rs (hunk 1 above)
Comment (elpiel) [reply src/with_prefix.rs:11]: I've used one of the existing prefixes in the example to do so.
Maybe you can be more specific if I have to create a full example like the Match - Player one below or just showcase a non-runnable example here?

Comment (jonasbb) [reply src/with_prefix.rs:11]: Just show a small non-runnable example of the syntax here. Similar to the sentence in the example. Such that you don't have to read the example to know that this is a possibility.

**Comment by jonasbb** [comment]
@elpiel I updated the description. Thanks again for creating the PR. I'm also releasing a new patch version.

**Comment by elpiel** [comment]
That's great, thank you!
Sorry, I didn't have much time to do it, I've been joggling between work and non-work projects 😅

**Pull request merged**